librosa/util/utils.py

```diff
@@ -35,80 +35,168 @@
+    x_frames : np.ndarray [shape=(..., frame_length, N_FRAMES) or (N_FRAMES, frame_length, ...)]
+        A framed view of `x`, for example with `axis=-1` (framing on the last dimension):
+        `x_frames[..., j] == x[..., j * hop_length : j * hop_length + frame_length]`
+
+        If `axis=0` (framing on the first dimension), then:
+        `x_frames[j] = x[j * hop_length : j * hop_length + frame_length]`
 
     Raises
     ------
     ParameterError
-        If `y` is not contiguous in memory, not an `np.ndarray`, or
-        not one-dimensional.  See `np.ascontiguous()` for details.
+        If `x` is not contiguous in memory or not an `np.ndarray`.
+
+        If `x.shape[axis] < frame_length`, there is not enough data to fill one frame.
 
         If `hop_length < 1`, frames cannot advance.
 
-        If `len(y) < frame_length`.
+        If `axis` is not 0 or -1.  Framing is only supported along the first or last axis.
+            If `axis=-1` (the default), then `x` must be "F-contiguous".
+            If `axis=0`, then `x` must be "C-contiguous".
+
+        If the contiguity of `x` is incompatible with the framing axis.
+
+    See Also
+    --------
+    np.asfortranarray : Convert data to F-contiguous representation
+    np.ascontiguousarray : Convert data to C-contiguous representation
+    np.ndarray.flags : information about the memory layout of a numpy `ndarray`.
 
     Examples
     --------
-    Extract 2048-sample frames from `y` with a hop of 64 samples per frame
+    Extract 2048-sample frames from monophonic `y` with a hop of 64 samples per frame
 
     >>> y, sr = librosa.load(librosa.util.example_audio_file())
-    >>> librosa.util.frame(y, frame_length=2048, hop_length=64)
-    array([[ -9.216e-06,   7.710e-06, ...,  -2.117e-06,  -4.362e-07],
-           [  2.518e-06,  -6.294e-06, ...,  -1.775e-05,  -6.365e-06],
+    >>> frames = librosa.util.frame(y, frame_length=2048, hop_length=64)
+    >>> frames
+    array([[ 0.000e+00,  0.000e+00, ..., -2.448e-06, -6.789e-07],
+           [ 0.000e+00,  0.000e+00, ..., -1.399e-05,  1.004e-06],
            ...,
-           [ -7.429e-04,   5.173e-03, ...,   1.105e-05,  -5.074e-06],
-           [  2.169e-03,   4.867e-03, ...,   3.666e-06,  -5.571e-06]], dtype=float32)
+           [-7.352e-04,  5.162e-03, ...,  0.000e+00,  0.000e+00],
+           [ 2.168e-03,  4.870e-03, ...,  0.000e+00,  0.000e+00]],
+          dtype=float32)
+    >>> y.shape
+    (1355168,)
+    >>> frames.shape
+    (2048, 21143)
+
+    Or frame along the first axis instead of the last:
+
+    >>> frames = librosa.util.frame(y, frame_length=2048, hop_length=64, axis=0)
+    >>> frames.shape
+    (21143, 2048)
 
+    Frame a stereo signal:
+
+    >>> y, sr = librosa.load(librosa.util.example_audio_file(), mono=False)
+    >>> y.shape
+    (2, 1355168)
+    >>> frames = librosa.util.frame(y, frame_length=2048, hop_length=64)
+    (2, 2048, 21143)
+
+    Carve an STFT into fixed-length patches of 32 frames with no overlap
```

Comment:
i would prefer an example with 50% overlap. For the user, getting it to work without overlap is simpler (`np.reshape`), as it does not involve `stride_tricks`. Does that make sense?